common-messaging/src/main/scala/org/genivi/sota/messaging/kafka/MessageListener.scala

```diff
@@ -15,33 +15,44 @@ object MessageListener {
 
   type MsgParser[T <: BusMessage] = T => Future[_]
 
-  def buildSource[T <: BusMessage](fromSource: Source[CommittableMessage[Array[Byte], T], NotUsed],
+  /**
+    * This trait is used to enable NATS to pretend it can do committable messages
+    */
+  trait CommittableMessageLike[T <: BusMessage] {
```

Comment:
Taking full responsibility for chosing the name `MessageLike`, I admit with was a bad name, but it meant to imply it's a type class, see the scala collection api for more on this.

But I think CommitableMessageLike is the wrong name, as this trait is just a trait, not a type class. When you call `commitRecord` you are calling it on a message instance, not on a type class.

So we have two options:

1. Implement real a type class, and suply that type class implicitly when we want to commit the record.

This has a nice property that the return type of subscribe commitable would stay `Source[T, NotUsed]`, except there would be an implicit value for `CommitableMessage[T]`.

2. Just change the name on this and use like it is now. What needs to change here is that this wrapping should be done at the client level, so KafkaClient source would return a CommitableMessage already, and this would be opaque to clients subscribing on each nats/kafka/local implementation.

I think the 2nd option might be easier to read and would complicate this less. Otherwise you'd have to supply different typeclasses depending on the client implementation and it's not immediate apparent to me how would I do this.